lib/puppet/provider/package/yum.rb

```diff
@@ -151,6 +154,36 @@ def self.update_command
     'update'
   end
 
+  def get_wanted_version(should)
+    if should =~ /[<>=~]/
+      begin
+        wanted_version = Puppet::Util::Package::Version::Range.parse(should, Version::Rpm)
+      rescue Puppet::Util::Package::Version::Range::ValidationFailure, Puppet::Util::Package::Version::Range::Rpm::ValidationFailure
+        raise Puppet::Error, _("Cannot parse %{should} as a RPM version") % { should: should }
+      end
+      # sort available versions in descending order
+      available_versions = get_available_versions(@resource[:name]).sort!.reverse!
+      for available_version in available_versions
```

Comment:
A more idiomatic way to do this in Ruby would be to use a block: `available_versions.each do |available_version|`